### PR TITLE
feat: Omit balanceOf requests for tokens that doesn't support it

### PR DESCRIPTION
**Pull request by Qwerty5Uiop**
## Motivation
There are tokens that doesn't support `balanceOf` function but we still keep and try to fetch balances for them which means that there would be endless errors on their fetching.

## Changelog
- Added table `missing_balance_of_token` that keeps information about such tokens and the max block number where `balanceOf` wasn't presented
- Filter token balances by this table before fetching
- Insert tokens for which `balanceOf` function returned `execution reverted`
- Filter token balances placeholders before insert

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/address/token_balance.ex [hunk 1]
```diff
@@ -119,4 +119,16 @@ defmodule Explorer.Chain.Address.TokenBalance do
...
+  Deletes all TokenBalances with given `token_contract_address_hash` and below the given `block_number`.
+  Used for cases when token doesn't implement balanceOf function
+  """
+  def delete_placeholders_below(token_contract_address_hash, block_number) do
+    TokenBalance
+    |> where([tb], tb.token_contract_address_hash == ^token_contract_address_hash)
+    |> where([tb], tb.block_number <= ^block_number)
+    |> where([tb], is_nil(tb.value_fetched_at) or is_nil(tb.value))
+    |> Repo.delete_all()
+  end
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/address/token_balance.ex:133]: Let's extract to shared function with addition module argument and re-use here and in CurrentTokenBalance module.

File: apps/explorer/priv/repo/migrations/20240502064431_create_missing_balance_of_tokens.exs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+defmodule Explorer.Repo.Migrations.CreateMissingBalanceOfTokens do
+  use Ecto.Migration
+
+  def change do
+    create table(:missing_balance_of_tokens, primary_key: false) do
+      add(:token_contract_address_hash, :bytea, primary_key: true)
+      add(:block_number, :bigint)
+    end
```
Comment (vbaranov) [inline apps/explorer/priv/repo/migrations/20240502064431_create_missing_balance_of_tokens.exs:8]: Let's add `timestamps()`?

File: apps/explorer/lib/explorer/chain/address/token_balance.ex (hunk 1 above)
Comment (Qwerty5Uiop) [reply apps/explorer/lib/explorer/chain/address/token_balance.ex]: Processed in 9295c69b7dc4b42840d3143b0e8c153c0c00727a

File: apps/explorer/priv/repo/migrations/20240502064431_create_missing_balance_of_tokens.exs (hunk 2 above)
Comment (Qwerty5Uiop) [reply apps/explorer/priv/repo/migrations/20240502064431_create_missing_balance_of_tokens.exs:10]: Processed in 9295c69b7dc4b42840d3143b0e8c153c0c00727a

**Review by vbaranov — APPROVED** [review]

**Pull request merged**